src/plugins/telemetry/server/routes/telemetry_usage_stats.ts

```diff
@@ -35,7 +35,7 @@ export function registerTelemetryUsageStatsRoutes(
         const statsConfig: StatsGetterConfig = {
           request: req,
           unencrypted,
-          refreshCache,
+          refreshCache: unencrypted || refreshCache,
```

Comment:
Q: Should this be handled down inside the `telemetryCollectionManager.getStats`? It may cause different behaviours if called from somewhere else. What do you think?

Reply:
I'd like to keep the logic at the route level since we might want to set up things differently in different places

Reply:
I see where you're coming from. However, the cache is implemented in `telemetryCollectionManager.getStats`, I don't see why others may want to use it in a different way. Especially bearing in mind that, `getStats` is the one deciding how to scope the requests (so any other place using `unencrypted: true, refreshCache: false` might expose again values retrieved with different permissions to the caller.

IMO, the scoping and the refresh algos should be together.